### DISCUSSION TITLE
Allow specifying custom header and content for custom amount section in PoS app

### DISCUSSION THREAD

**Discussion by bolatovumar** [btcpayserver/btcpayserver#2431] [Ideas / Feature Requests]
I have a donation site [here](https://donate.umar-bolatov.com) where I only have the custom amount option. It would be nice to be able to specify custom content there instead of the default content. So, instead of "Custom Amount" I could say something like "gib sats plz" and instead of "Create invoice to pay custom amount" I could say "btcpay 4eva".

![Screen Shot 2021-04-04 at 1 58 50 PM](https://user-images.githubusercontent.com/1934678/113521385-06dbfc00-954e-11eb-9e87-335b5a0234f9.png)

**Comment by ghost** [comment]
Yes, I would like this too. 
cACK, would use in production.

**Comment by Kukks** [comment]
This is already possible. IMO, the custom option is very redundant with how flexible the options have become now and could be removed.

![image](https://user-images.githubusercontent.com/1818366/113553693-6c84c280-95f8-11eb-9e24-74c811d93158.png)
![image](https://user-images.githubusercontent.com/1818366/113553718-74dcfd80-95f8-11eb-81fb-554e7dac2e76.png)

**Reply by Zaxounette** [reply]
I concur, this feature request simply amounts to using the POS app with a single item for which you enable the custom price.

**Reply by bolatovumar** [reply]
Yeah, I guess with the recent changes this can be done already. I will look into it to make sure we are not missing anything here. If nothing is missing, I will just close this.

**Reply by bolatovumar** [reply]
Ok, I created a store with both options side by side:

![Screen Shot 2021-04-08 at 8 47 25 PM](https://user-images.githubusercontent.com/1934678/114125620-b5b06d00-98ab-11eb-87f0-6ac7f399deae.png)

Looks like you can accomplish what I'm after with just a regular store item. There are some subtle but important differences, however:

1. The button width is not the same. The custom amount version is nicer as it doesn't stretch the button. I wonder why it's done this way for regular store items. (nvm, this doesn't apply if the regular store item is the ONLY item in the store)
2. The custom amount version has placeholder text in the input (Amount) instead of just showing you a number. This is nice. The other version, however, allows you specify a suggested amount which is also nice.
3. I think there is a bug with the regular store item where if you leave the default price of 0 on, clicking on the pay button will create an invoice for 0 amount and it will be marked as paid right away. This should probably not happen.

**Reply by Kukks** [reply]
1) This looks like a bug. We should make them identical
2) Price is mandatory, which is why this happens. cACK to make it optional for items marked with custom
3) Feature not bug, allows you to use pos to distribute free items. E.g. Limited amount of free tickets to an event

**Reply by bolatovumar** [reply]
@Kukks created a PR to address issue 1 here: https://github.com/btcpayserver/btcpayserver/pull/2453
